torchao/quantization/prototype/qat/utils.py

```diff
@@ -29,19 +38,21 @@ class _GenericFakeQuantize(torch.autograd.Function):
     def forward(
         ctx: torch.autograd.function.FunctionCtx,
         input: torch.Tensor,
+        block_size: List[int],
         scales: torch.Tensor,
         zero_points: torch.Tensor,
         quant_min: int,
         quant_max: int,
-        block_size: List[int],
         zero_point_domain: ZeroPointDomain = ZeroPointDomain.INT,
     ) -> torch.Tensor:
-        # Note: for bf16 inputs, casting them to fp32 has the unexpected
-        # side effect of reducing memory footprint significantly, presumably
-        # because bf16 * fp32 kernels are not as memory efficient
-        assert input.dtype == torch.float32
-        assert scales.dtype == torch.float32
-        assert zero_points.dtype == torch.int32
+        # avoid circular dependencies
+        from torchao.quantization.prototype.qat.affine_fake_quantized_tensor import (
+            AffineFakeQuantizedTensor,
+        )
+
+        original_input = input
+        if isinstance(original_input, AffineFakeQuantizedTensor):
+            input = input.original_tensor
```

Comment:
why don't we pass in original_tensor directly here?

Reply:
We're passing in the `original_tensor` here. Maybe the naming is confusing? Let me rewrite this